### PR TITLE
DEV-3403 Introduce until to the client

### PR DESCRIPTION
**Pull request by zsid60**
The relevant changes are in `to_proto.py` and in `sources/`
All other changes are linter related or proto regens.
Protos are regenerated off of https://github.com/fennel-ai/server/pull/1225

Docs
image

image

### CODE REVIEW

File: fennel/sources/sources.py [hunk 1]
```diff
@@ -56,9 +57,16 @@ def source(
     if since is not None and not isinstance(since, datetime):
         raise TypeError(f"'since' must be of type datetime - got {type(since)}")
 
+    if until is not None and not isinstance(until, datetime):
+        raise TypeError(f"'until' must be of type datetime - got {type(until)}")
+
+    if since is not None and until is not None and since > until:
+        raise ValueError("'since' must be before 'until'")
```
Comment (mohitreddy1996) [inline fennel/sources/sources.py:64]: we should also log what source is and until is? just to have a complete error message?

Comment (mohitreddy1996) [inline fennel/sources/sources.py:64]: maybe `earlier than` instead of `before`? Makes the fields in the error message feel more time dimensional based.

**Review by mohitreddy1996 — APPROVED** [review]
please do wait for @nikhilgarg28 review on documentation

**Comment by mohitreddy1996** [comment]
@zsid60 can you ping @nikhilgarg28 for a quick review?

**Comment by nikhilgarg28** [comment]
Docs changes look good.

**Pull request merged**